diff --git a/source3/smbd/server.c b/source3/smbd/server.c
index 8c77500..fa05a28 100644
--- a/source3/smbd/server.c
+++ b/source3/smbd/server.c
@@ -1200,7 +1200,7 @@ extern void build_options(bool screen);
 	}
 
 	if (print_build_options) {
-		build_options(True); /* Display output to screen as well as debug */
+		DEBUG(0, ("For build options, please check the Debian packaging (apt-get source samba)."));
 		exit(0);
 	}
 
@@ -1271,9 +1271,6 @@ extern void build_options(bool screen);
 	DEBUG(2,("uid=%d gid=%d euid=%d egid=%d\n",
 		 (int)getuid(),(int)getgid(),(int)geteuid(),(int)getegid()));
 
-	/* Output the build options to the debug log */ 
-	build_options(False);
-
 	if (sizeof(uint16_t) < 2 || sizeof(uint32_t) < 4) {
 		DEBUG(0,("ERROR: Samba is not configured correctly for the word size on your machine\n"));
 		exit(1);
diff --git a/source3/wscript_build b/source3/wscript_build
index 8adb942..fbf79c8 100755
--- a/source3/wscript_build
+++ b/source3/wscript_build
@@ -7,16 +7,6 @@ LIBS='ICONV'
 
 config_h = "../include/config.h"
 
-bld.SAMBA_BLDOPTIONS('smbd/build_options.c')
-
-t = bld.SAMBA_GENERATOR('build_env.h',
-                        source='script/build_env.sh',
-                        target='include/build_env.h',
-                        rule='${SRC} ${SRCDIR} ${BUILDDIR} ${CC} > ${TGT}')
-# todo: work out what is really wanted here
-t.env.SRCDIR = bld.path.abspath()
-t.env.BUILDDIR = bld.path.abspath()
-
 bld.SETUP_BUILD_GROUPS()
 
 ######################## SUBSYSTEMS #################################
@@ -609,7 +599,7 @@ bld.SAMBA3_LIBRARY('smbd_base',
                    smbd/oplock_linux.c
                    smbd/notify.c
                    smbd/notify_msg.c
-                   smbd/build_options.c''' + NOTIFY_SOURCES,
+                   ''' + NOTIFY_SOURCES,
                    deps='''
                    talloc
                    tevent
